### ISSUE TITLE
rust-analyzer analysis-stats crash with proc macro

### ISSUE THREAD

**Issue by pacak** [rust-analyzer/rust-analyzer#9414]
```
% rust-analyzer --version
rust-analyzer 3898387f3 2021-06-21 stable 
```

```
Database loaded:     605.66ms, 50minstr
thread 'main' panicked at 'range end index 8 out of range for slice of length 7', crates/proc_macro_srv/src/proc_macro/bridge/rpc.rs:135:1
stack backtrace:
   0: rust_begin_unwind
             at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b/library/std/src/panicking.rs:493:5
   1: core::panicking::panic_fmt
             at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b/library/core/src/panicking.rs:92:14
   2: core::slice::index::slice_end_index_len_fail
             at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b/library/core/src/slice/index.rs:41:5
   3: <core::ops::range::Bound<T> as proc_macro_srv::proc_macro::bridge::rpc::DecodeMut<S>>::decode
   4: <std::panic::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
   5: <proc_macro_srv::proc_macro::bridge::server::Dispatcher<proc_macro_srv::proc_macro::bridge::server::MarkedTypes<S>> as proc_macro_srv::proc_macro::bridge::server::DispatcherTrait>::dispatch
   6: <proc_macro_srv::proc_macro::bridge::closure::Closure<A,R> as core::convert::From<&mut F>>::from::call
   7: proc_macro::bridge::closure::Closure<A,R>::call
             at /rustc/e4ca1662f255db774dcd16ed9c3776bf25ac8361/library/proc_macro/src/bridge/closure.rs:27:18
   8: proc_macro::bridge::client::Literal::set_span::{{closure}}
             at /rustc/e4ca1662f255db774dcd16ed9c3776bf25ac8361/library/proc_macro/src/bridge/client.rs:244:25
   9: proc_macro::bridge::client::<impl proc_macro::bridge::Bridge>::with::{{closure}}
             at /rustc/e4ca1662f255db774dcd16ed9c3776bf25ac8361/library/proc_macro/src/bridge/client.rs:336:47
  10: proc_macro::bridge::client::BridgeState::with::{{closure}}::{{closure}}
....
```

[crash.log](https://github.com/rust-analyzer/rust-analyzer/files/6723080/crash.log)
[repro.tar.gz](https://github.com/rust-analyzer/rust-analyzer/files/6723082/repro.tar.gz)

**Comment by lnicola** [comment]
Are you on a nightly rustc?

**Comment by pacak** [comment]
yes

**Comment by bjorn3** [comment]
Duplicate of https://github.com/rust-analyzer/rust-analyzer/issues/8925

tl;dr: rustc nightly changed the unstable proc macro abi. rust-analyzer can't adopt to the changes until the changes land on stable as otherwise proc macros break on stable.
